File: tests/fixtures/sample.go
// Sample Go file for testing
package main

import (
	"fmt"
	"strings"
)

// MaxUsers is the maximum number of users allowed
const MaxUsers = 1000

// DefaultName is the default user name
const DefaultName = "Anonymous"

// globalCounter tracks the total number of users created
var globalCounter int

// User represents a user in the system
type User struct {
	ID    int
	Name  string
	Email string
}

// UserService manages user operations
type UserService struct {
	users []User
}

// Validatable is an interface for entities that can be validated
type Validatable interface {
	IsValid() bool
	Validate() error
}

// UserID is a type alias for user identifiers
type UserID = int

// NewUser creates a new User instance
func NewUser(id int, name, email string) *User {
	globalCounter++
	return &User{
		ID:    id,
		Name:  name,
		Email: email,
	}
}

// DisplayName returns the user's display name
func (u *User) DisplayName() string {
	if u.Name == "" {
		return DefaultName
	}
	return u.Name
}

// IsValid checks if the user is valid
func (u *User) IsValid() bool {
	return u.Name != "" && strings.Contains(u.Email, "@")
}

// Validate validates the user and returns an error if invalid
func (u *User) Validate() error {
	if u.Name == "" {
		return fmt.Errorf("name cannot be empty")
	}
	if !strings.Contains(u.Email, "@") {
		return fmt.Errorf("invalid email format")
	}
	return nil
}

// NewUserService creates a new UserService
func NewUserService() *UserService {
	return &UserService{
		users: make([]User, 0),
	}
}

// AddUser adds a user to the service
func (s *UserService) AddUser(user User) {
	s.users = append(s.users, user)
}

// FindUser finds a user by ID
func (s *UserService) FindUser(id int) *User {
	for i := range s.users {
		if s.users[i].ID == id {
			return &s.users[i]
		}
	}
	return nil
}

// processUser processes a user
func processUser(user *User) string {
	return fmt.Sprintf("Processing user: %s", user.DisplayName())
}

// validateEmail validates an email address
func validateEmail(email string) bool {
	return strings.Contains(email, "@") && strings.Contains(email, ".")
}

// TODO: Add more validation rules
// FIXME: Handle edge cases for empty strings

func main() {
	// Create a new user
	user := NewUser(1, "John Doe", "john@example.com")
	fmt.Println(user.DisplayName())

	// Create a user service
	service := NewUserService()
	service.AddUser(*user)

	// Process the user
	result := processUser(user)
	fmt.Println(result)
}

File: tests/fixtures/sample.rs
// Sample Rust file for testing

use std::collections::HashMap;
use std::fmt::Display;

/// A user in the system
pub struct User {
    pub id: u64,
    pub name: String,
    pub email: String,
}

impl User {
    /// Creates a new user
    pub fn new(id: u64, name: String, email: String) -> Self {
        Self { id, name, email }
    }

    /// Gets the user's display name
    pub fn display_name(&self) -> &str {
        &self.name
    }
}

impl Display for User {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "User {{ id: {}, name: {} }}", self.id, self.name)
    }
}

/// User role enumeration
pub enum UserRole {
    Admin,
    Moderator,
    Member,
    Guest,
}

impl UserRole {
    /// Checks if the role has admin privileges
    pub fn is_admin(&self) -> bool {
        matches!(self, UserRole::Admin)
    }
}

/// A trait for entities that can be validated
pub trait Validatable {
    /// Validates the entity
    fn validate(&self) -> Result<(), String>;

    /// Checks if the entity is valid
    fn is_valid(&self) -> bool {
        self.validate().is_ok()
    }
}

impl Validatable for User {
    fn validate(&self) -> Result<(), String> {
        if self.name.is_empty() {
            return Err("Name cannot be empty".to_string());
        }
        if !self.email.contains('@') {
            return Err("Invalid email format".to_string());
        }
        Ok(())
    }
}

/// Type alias for user ID
pub type UserId = u64;

/// Type alias for a user map
pub type UserMap = HashMap<UserId, User>;

/// User service module
pub mod user_service {
    use super::*;

    /// Service for managing users
    pub struct UserService {
        users: UserMap,
    }

    impl UserService {
        /// Creates a new user service
        pub fn new() -> Self {
            Self {
                users: HashMap::new(),
            }
        }

        /// Adds a user to the service
        pub fn add_user(&mut self, user: User) {
            self.users.insert(user.id, user);
        }

        /// Gets a user by ID
        pub fn get_user(&self, id: UserId) -> Option<&User> {
            self.users.get(&id)
        }
    }

    impl Default for UserService {
        fn default() -> Self {
            Self::new()
        }
    }
}

/// Maximum number of users
pub const MAX_USERS: usize = 1000;

/// Default user name
pub const DEFAULT_NAME: &str = "Anonymous";

/// Global user counter
pub static mut USER_COUNTER: u64 = 0;

/// Increments and returns the user counter
///
/// # Safety
/// This function is unsafe because it mutates a static variable
pub unsafe fn next_user_id() -> u64 {
    USER_COUNTER += 1;
    USER_COUNTER
}

/// Helper function to process a user
pub fn process_user(user: &User) -> String {
    format!("Processing user: {}", user.display_name())
}

/// Helper function to validate email
pub fn validate_email(email: &str) -> bool {
    email.contains('@') && email.contains('.')
}

/// A macro for creating users
macro_rules! create_user {
    ($id:expr, $name:expr, $email:expr) => {
        User::new($id, $name.to_string(), $email.to_string())
    };
}

// TODO: Add more validation
// FIXME: Handle edge cases

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_user_creation() {
        let user = create_user!(1, "John", "john@example.com");
        assert_eq!(user.id, 1);
        assert_eq!(user.name, "John");
    }
}
